src/wp-content/themes/twentytwelve/functions.php

```diff
@@ -415,10 +415,25 @@ function twentytwelve_content_nav( $html_id ) {
 		global $wp_query;
 
 		if ( $wp_query->max_num_pages > 1 ) : ?>
+			<?php
+			$order   = get_query_var( 'order', 'DESC' );
+			$is_desc = ( 'DESC' === $order );
+
+			$new_posts_text = __( 'Newer posts <span class="meta-nav">&rarr;</span>', 'twentytwelve' );
+			$old_posts_text = __( '<span class="meta-nav">&larr;</span> Older posts', 'twentytwelve' );
+
+			$prev_link = $is_desc ? get_next_posts_link( $old_posts_text ) : get_previous_posts_link( $old_posts_text );
+			$next_link = $is_desc ? get_previous_posts_link( $new_posts_text ) : get_next_posts_link( $new_posts_text );
+			?>
 			<nav id="<?php echo esc_attr( $html_id ); ?>" class="navigation">
 				<h3 class="assistive-text"><?php _e( 'Post navigation', 'twentytwelve' ); ?></h3>
-				<div class="nav-previous"><?php next_posts_link( __( '<span class="meta-nav">&larr;</span> Older posts', 'twentytwelve' ) ); ?></div>
-				<div class="nav-next"><?php previous_posts_link( __( 'Newer posts <span class="meta-nav">&rarr;</span>', 'twentytwelve' ) ); ?></div>
+				<?php if ( $prev_link ) : ?>
+					<div class="nav-previous"><?php echo wp_kses_post( $prev_link ); ?></div>
+				<?php endif; ?>
+
+				<?php if ( $next_link ) : ?>
+					<div class="nav-next"><?php echo wp_kses_post( $next_link ); ?></div>
+				<?php endif; ?>
 			</nav><!-- .navigation -->
 			<?php
 	endif;
```

Comment:
```suggestion
		endif;
```

The `endif` can align with the `if` statement.